tracer/tracer.go

```diff
@@ -1103,16 +1104,20 @@ func (t *Tracer) AttachTracer() error {
 		return fmt.Errorf("failed to configure software perf event: %v", err)
 	}
 
-	onlineCPUIDs, err := hostcpu.ParseCPUCoreIDs(hostcpu.CPUOnlinePath)
+	possibleCPUs, err := numcpus.GetPossible()
```

Comment:
We should probably keep the previous method of retrieving online CPUs by parsing the path in `/sys/devices` and then proceeding with perf calls only on those CPUs instead of running `perf.Open` on the entire possible CPU range.

Reply:
What is the concern here? With `numcpus.GetPossible()` we get the maximum number of possible CPUs in the system - Online CPUs ≤ Present CPUs ≤ Possible CPUs.
Iterating over all possible CPUs and trying to attach a perf event might be slower, than iterating over all online CPUs to attach perf events - if possible CPUs != online CPUs.
With ENODEV there is a well established error code returned by the perf subsystem for cases, where a perf event can not be attached.
The proposed approach is less error prone and easier to follow. E.g. the content of `/sys/devices/system/cpu/online` could look like `0-7,9,13,21-31` for a system with 32 possible CPUs. Parsing this content, generating the correct list of all online CPUs is more error prone that iterating over 31 possible CPUs.

Reply:
Parsing the file should result in a single system call, while iterating over all possible CPUs could result in thousands of `perf_event_open` syscalls that are guaranteed to fail as the possible CPU range on average will be much larger than the online or present CPU range. That said, I didn't benchmark and I don't expect the runtime hit to be significant with the possible exception of an EDR that installs hooks.